vega_sim/scenario/network/scenario.py

```diff
@@ -0,0 +1,136 @@
+"""network.py
+
+Module for running a predefined scenario on a Vega network.
+"""
+
+import os
+import dotenv
+import argparse
+import logging
+from random import random
+import numpy as np
+from typing import Any, Callable, List, Optional
+from vega_sim.environment.agent import Agent
+
+from vega_sim.scenario.scenario import Scenario
+from vega_sim.environment.environment import NetworkEnvironment
+from vega_sim.network_service import VegaServiceNetwork
+
+from vega_sim.scenario.common.agents import (
+    MarketOrderTrader,
+    LimitOrderTrader,
+)
+
+
+class Fairground(Scenario):
+    def __init__(
+        self,
+        num_steps: int = 5,
+        step_length_seconds: int = 1,
+        market_name: str = None,
+    ):
+
+        dotenv.load_dotenv()
+        self.wallet_name = os.environ.get("VEGA_USER_WALLET_NAME")
+        self.wallet_pass = os.environ.get("VEGA_USER_WALLET_PASS")
+
+        self.num_steps = num_steps
+        self.step_length_seconds = step_length_seconds
+
+        self.market_name = market_name
+        self.market_id = None
+        self.asset_name = None
+
+        self.random_state = np.random.RandomState()
+
+    def set_up_background_market(
+        self,
+        vega: VegaServiceNetwork,
+        tag: str = "",
+        random_state: Optional[np.random.RandomState] = None,
+    ) -> NetworkEnvironment:
+
+        self.market_id = [
+            m.id
+            for m in vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+        market_info = vega.market_info(market_id=self.market_id)
+        self.asset_name = (
+            market_info.tradable_instrument.instrument.future.settlement_asset
+        )
+
```

Comment:
I assume currently if one runs this all the traders try to trade with the same key and cause chaos? 😄

Reply:
Yep exactly 😆 Now the multi-key wallets PR is merged I'll upgrade this scenario.